wsf_scraping/spiders/base_spider.py

```diff
@@ -28,24 +26,58 @@ def on_error(self, failure):
             request = failure.request
             self.logger.error('TimeoutError on %s', request.url)
 
+        else:
+            self.logger.error(repr(failure))
+
     def _check_headers(self, response_headers,
                        desired_extension=b'application/pdf'):
         content_type = response_headers.get('content-type', '').split(b';')[0]
         return desired_extension == content_type
 
-    def _save_file(self, url, response_body):
-        filename = os.path.basename(urlparse(url).path)
-        tempdir = tempfile.mkdtemp()
-        if filename:
-            if not filename.lower().endswith('.pdf'):
-                filename = filename + '.pdf'
-            filepath = os.path.join(tempdir, filename)
-            with open(filepath, 'wb') as f:
-                f.write(response_body)
-            return filepath
-        else:
+    def save_pdf(self, response):
+        """ Save the response body to a temporary PDF file.
+
+        If the response body is PDF-typed, save the PDF to a tempfile to parse
+        it later. Else, just drop te item.
+
+        Args:
+            - response: The reponse object passed by scrapy
+
+        Returns:
+            - A scrapy Article item.
+        """
+
+        data_dict = response.meta.get('data_dict', {})
+
+        is_pdf = self._check_headers(response.headers)
+
+        if not is_pdf:
+            self.logger.info('Not a PDF, aborting (%s)', response.url)
+            return
+
+        if not response.body:
             self.logger.warning(
-                 'Empty filename, could not save the file. [Url: %s]',
-                 url
+                'Empty filename or content, could not save the file.'
+                ' [Url: %s]',
+                response.request.url
             )
-            return ''
+            return
+
+        # Download PDF file to /tmp
+        with tempfile.NamedTemporaryFile(delete=False) as tf:
+            tf.write(response.body)
```

Comment:
It'd be good to note here whose job it will be to delete the file.

Reply:
Done